src/crawlee/base_storage_client/_base_dataset_client.py

```diff
@@ -122,6 +122,11 @@ async def iterate_items(
             An asynchronous iterator of dictionary objects, each representing a dataset item after applying
             the specified filters and transformations.
         """
+        # This syntax is to make mypy properly work with abstract AsyncIterator.
+        # https://mypy.readthedocs.io/en/stable/more_types.html#asynchronous-iterators
+        raise NotImplementedError
+        if False:
```

Comment:
Unfortunately, this will be necessary

```suggestion
        if False:  # type: ignore
```